rasa_sdk/executor.py

```diff
@@ -23,49 +23,69 @@ def __init__(self) -> None:
 
         self.messages = []
 
+    def utter_message(
+        self,
+        image: Optional[Text] = None,
+        json_message: Dict[Text, Any] = None,
+        template: Optional[Text] = None,
+        attachment: Optional[Text] = None,
+        text: Optional[Text] = None,
+        buttons: Optional[List[Dict[Text, Any]]] = None,
+        *elements: Dict[Text, Any],
+        **kwargs: Any,
+    ) -> None:
+        """"Send a text to the output channel"""
+
+        message = {
+            "text": text,
+            "buttons": buttons,
+            "elements": elements,
+            "custom": json_message,
+            "template": template,
+            "image": image,
+            "attachment": attachment,
+        }
+        message.update(kwargs)
+
+        self.messages.append(message)
+
     # deprecated
     def utter_custom_message(self, *elements: Dict[Text, Any], **kwargs: Any) -> None:
         warnings.warn(
             "Use of `utter_custom_message` is deprecated. "
-            "Use `utter_elements` to send elements, or "
-            "`utter_custom_json` to send a custom json message. ",
+            "Use `utter_message` instead.",
             DeprecationWarning,
         )
         self.utter_elements(*elements, **kwargs)
 
     def utter_elements(self, *elements: Dict[Text, Any], **kwargs: Any) -> None:
         """Sends a message with custom elements to the output channel."""
-
-        message = {"text": None, "elements": elements}
-        message.update(kwargs)
-
-        self.messages.append(message)
-
-    def utter_message(self, text: Text, **kwargs: Any) -> None:
-        """"Send a text to the output channel"""
-
-        message = {"text": text}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        warnings.warn(
+            "Use of `utter_elements` is deprecated. " "Use `utter_message` instead.",
+            DeprecationWarning,
+        )
+        self.utter_message(*elements)
 
     def utter_button_message(
         self, text: Text, buttons: List[Dict[Text, Any]], **kwargs: Any
     ) -> None:
         """Sends a message with buttons to the output channel."""
+        warnings.warn(
+            "Use of `utter_button_message` is deprecated. "
+            "Use `utter_message` instead.",
+            DeprecationWarning,
+        )
 
-        message = {"text": text, "buttons": buttons}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        self.utter_message(text=text, buttons=buttons)
```

Comment:
What about `kwargs`? Same for the other methods.

Reply:
Thanks for catching that 👍 I will pass on the kwargs
(Out of curiosity: do you know if they are ever used?)

Reply:
I have no idea 😄

Reply:
better safe than sorry 😂 
just made the change